### ISSUE TITLE
Flaky test: "Testing bswap i32 for 13 elements"

### ISSUE THREAD

**Issue by dvzrv** [sadko4u/lsp-plugins#19]
When building 1.1.4 and also 1.1.5 I *sometimes* run into flaky tests:

```
Testing bswap i32 for 12 elemens
TestUnit test 'dsp.3d.triplet' assertion has failed at file utest/dsp/3d/triplet.cpp, line 94:
  check_triplet3d_v2n(&v[1], &v[2], &n1) >= 0
ing bswap i32 for 13 elemens
Testing bswap i32 for 14 elemens
```

I run the tests by first doing a separate test build, then running the tests only on that:

```
  local _test_path="$(pwd)/.build-test"
  cd "$pkgname-$pkgver"
  make
  make OBJDIR=${_test_path} clean
  make OBJDIR=${_test_path} test
  local _test_path="$(pwd)/.build-test"
  cd "$pkgname-$pkgver"
  ${_test_path}/lsp-plugins-test utest --nofork --debug --verbose
```

It puzzles me, that I only run into that failing test ~ every 5th time I rebuild...
Maybe you have any ideas or have encountered similar problems?
It's pretty bad for reproducible builds...

**Comment by dvzrv** [comment]
Hm... and now I'm experiencing the test failing every time... this is getting weird. :-/

**Comment by sadko4u** [comment]
What's the final list of failed tests (it is listed after all tests are executed)?
What's the architecture and platform?
BSwap should not fail but 'dsp.3d.triplet' can blink, it depends on the floating-point subsystem of machine.

**Comment by sadko4u** [comment]
I'm having no problem on my machine (AMD Ryzen 2700):
```
--------------------------------------------------------------------------------
Overall unit test statistics:
  overall time [s]:     2.76
  launched:             64
  succeeded:            64
  failed:               0
```

**Comment by sadko4u** [comment]
Yes, I see a blinking test:
```
TestUnit test 'dsp.3d.triplet' assertion has failed at file utest/dsp/3d/triplet.cpp, line 94:
  check_triplet3d_v2n(&v[1], &v[2], &n1) >= 0
```
I think i should rework this test and add possibility to manually ignore some tests from command-line.
The problem is that I don't see the overall statistics in your log. For example, if two tests fail, I get at the end:
```
--------------------------------------------------------------------------------
Overall unit test statistics:
  overall time [s]:     2.80
  launched:             64
  succeeded:            62
  failed:               2

--------------------------------------------------------------------------------
List of failed tests:
  dsp.3d.triplet
  dsp.graphics.axis
```

**Comment by sadko4u** [comment]
Also, one another moment: because unit tests utilize all CPU cores when launching, you may misinterpret the output and consider that another unit test has failed. The real list of failed tests is exactly in 'List of failed tests' section. All other is just log output.

**Comment by dvzrv** [comment]
Hm, that's all the output I get from running the unit tests though. Is there anything I'm doing wrong in running them? I followed your suggestions (about the 2nd build) and all from one of the other issues recently.

**Comment by sadko4u** [comment]
With this commit, I've added possibility to ignore some unit tests: f0f9c9e68b5717bcc03fe7d2771d21a3bcf8aff6

You can ignore the flaky test by specifying it's name in the command line after **-i** switch:
```
.build/lsp-plugins-test utest -i dsp.3d.triplet
```

Also please notice that this test does not perform unit-testing of routines used in production, so you can surely disable it for the 1.1.5 release. I believe in 1.1.6 it will be reworked or, even more, removed.

**Comment by sadko4u** [comment]
This commit has finally fixed the problem: 49abe134ea4827248edfb4404214a0cbef1fc88b

**Comment by sadko4u** [comment]
Merged changes into *devel* branch: b896c50e5c34734ff2eadf33834553aa3a6e7f19
